File: 20181124-java-fibonacci/src/test/java/com/katas/FibonacciSequenceTest.java
package com.katas;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class FibonacciSequenceTest {
    @Test
    public void first_number_is_zero() {
        FibonacciSequence fibonacci = new FibonacciSequence();

        assertThat(fibonacci.getNumber(0)).isEqualTo(0);
    }
}

File: 20210323-kotlin-fibonacci/src/test/kotlin/com/katas/FibonacciSequenceTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class FibonacciSequenceTest {

    private val fibonacciSequence = FibonacciSequence()

    @Test
    fun `first number in sequence is zero`() {
        assertThat(fibonacciSequence.numberAt(0)).isZero()
    }

    @Test
    fun `second number in sequence is one`() {
        assertThat(fibonacciSequence.numberAt(1)).isOne()
    }

    @ParameterizedTest
    @CsvSource(
        "2, 1",
        "3, 2",
        "5, 5"
    )
    fun `other numbers are the sum of previous two`(position: Int, number: Int) {
        assertThat(fibonacciSequence.numberAt(position)).isEqualTo(number)
    }
}
